File: euler7.rb
def primality(n)
	sqrt = Integer.sqrt(n)
	for i in 2..sqrt
		if n % i == 0
			return 0
		end
	end
return n
end


count = 2
for i in 2..105000
	if count == 10001
		puts i-1
		break
	elsif primality(i) == i
		count+=1
	end
end

File: euler7.c
#include<stdio.h>
#include<math.h>

int is_prime(int n);


int main()
{
  int count = 1;
  for(int x = 2 ; x < 1000000; x++)
  {
	if ( count == 10001)
	{
		printf("The 10,001st prime is %d\n",x);
		break;
	}
	if ( is_prime(x) == x)
	{
		count+=1;
	}
  }
}

int is_prime(int n)
{
	int a = sqrt(n);
	for(int x = 2; x <= a; x++)
	{
		if ( n % x == 0)
		{
			return 0;
		}
	}
	return n;
}
